demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,505 @@
+                // limit threading for CPU portion of inference
+                if (FLAGS_nthreads != 0)
+                    ie.SetConfig({{ CONFIG_KEY(CPU_THREADS_NUM), std::to_string(FLAGS_nthreads) }}, device);
+
+                if ((FLAGS_d.find("MULTI") != std::string::npos) &&
+                    (FLAGS_d.find("GPU") != std::string::npos)) {
+                    ie.SetConfig({{ CONFIG_KEY(CPU_BIND_THREAD), CONFIG_VALUE(NO) }}, device);
+                } else {
+                    // pin threads for CPU portion of inference
+                    ie.SetConfig({{ CONFIG_KEY(CPU_BIND_THREAD), "YES" }}, device);
+                }
+
+                // for CPU execution, more throughput-oriented execution via streams
+                ie.SetConfig({{ CONFIG_KEY(CPU_THROUGHPUT_STREAMS),
+                                (deviceNstreams.count(device) > 0 ? std::to_string(deviceNstreams.at(device)) :
+                                                                                 "CPU_THROUGHPUT_AUTO") }}, device);
+                deviceNstreams[device] = std::stoi(
+                    ie.GetConfig(device, CONFIG_KEY(CPU_THROUGHPUT_STREAMS)).as<std::string>());
+            } else if (device == "GPU") {
+                ie.SetConfig({{ CONFIG_KEY(GPU_THROUGHPUT_STREAMS),
+                                (deviceNstreams.count(device) > 0 ? std::to_string(deviceNstreams.at(device)) :
+                                                                         "GPU_THROUGHPUT_AUTO") }}, device);
+                deviceNstreams[device] = std::stoi(
+                    ie.GetConfig(device, CONFIG_KEY(GPU_THROUGHPUT_STREAMS)).as<std::string>());
+
+                if ((FLAGS_d.find("MULTI") != std::string::npos) &&
+                    (FLAGS_d.find("CPU") != std::string::npos)) {
+                    // multi-device execution with the CPU + GPU performs best with GPU throttling hint,
+                    // which releases another CPU thread (that is otherwise used by the GPU driver for active polling)
+                    ie.SetConfig({{ CLDNN_CONFIG_KEY(PLUGIN_THROTTLE), "1" }}, "GPU");
+                }
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // --------------------------------------Load network to device---------------------------------------
+        ExecutableNetwork executableNetwork = ie.LoadNetwork(network, FLAGS_d);
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------Try to set optimal number of infer requests----------------------------
+        if (FLAGS_nireq == 0) {
+            std::string key = METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS);
+            try {
+                FLAGS_nireq = executableNetwork.GetMetric(key).as<unsigned int>();
```

Comment:
Flag value shouldn't be modified because it represents the input. Better introduce another var.